### ISSUE TITLE
rolebindings gets overwritten by upgrade.sh

### ISSUE THREAD

**Issue by mjudeikis** [jim-minter/azure-helm#94]
How to recreate:

1. Create a cluster
2. `oc adm policy add-cluster-role-to-user cluster-admin demo`
3. Run upgrade.sh
4. Enjoy lost privileges.

**Comment by mjudeikis** [comment]
forgot we have this stuff now :D

**Comment by kargakis** [comment]
Labels in our repo are still not synced, kicked the label sync job.

**Comment by kargakis** [comment]
Yeah, today it works as designed but we probably need a different way to
reconcile RBAC and SCC updates. There is actually a command for reconciling
these that is meant to be used on upgrades.

**Comment by mjudeikis** [comment]
@jim-minter @kargakis  this is the idea for the ticket. It works as expected now, but it not fit for RBAC and SCC as it is. So we will need slightly different way to sync these kinds of objects.

**Comment by jim-minter** [comment]
Really? I'm supposing that for now as part of our engineering effort, we would reconcile ourselves, update the sync definitions to what we want to them to be, then sync sets them correctly on the remote end,

**Comment by kargakis** [comment]
I think we don't overwrite user data on any other type of deployment during
upgrades so we may need to document this and it may be fine at least for PP.

**Comment by mjudeikis** [comment]
for GA: User configures Azures AAD to say `Group1` is my `cluster-admin`, `Group2` is my `view`.  If this will be a case we can work this out in our sync plugin.
for PP: if we say we don't have group integration, user logins with his account (does not matter AAD or HTPasswd) and grants `user1`, `user2` `cluster-admin` for purpose of PP work. Now our sync plugin runs and this particular `RoleBinding` is overwritten. 

We either need to find a way how to do these things and document or follow how users are used to doing things now.

**Comment by jim-minter** [comment]
Users and Groups are specifically excluded by the sync code (everything under user.openshift.io).  At least to begin with we are not allowing users cluster-admin, nor are we letting them create ClusterRoleBindings.

**Comment by kargakis** [comment]
> Users and Groups are specifically excluded by the sync code (everything under user.openshift.io). At least to begin with we are not allowing users cluster-admin, nor are we letting them create ClusterRoleBindings.

Good enough, @mjudeikis close?

**Comment by mjudeikis** [comment]
> Users and Groups are specifically excluded by the sync code (everything under user.openshift.io). At least to begin with we are not allowing users cluster-admin, nor are we letting them create ClusterRoleBindings.

I might be missing some story here. I'm a user/organization. I just provisioned a cluster and want to grant `John` access to the my project, which I `oc new-project` ? Or my Azure AD group access to all projects? This looks like one of the main requirements to access cluster? Or I'm missing something here?

**Comment by kargakis** [comment]
Assigning access to rolebindings (namespace-scoped) should not be affected by the sync process. @mjudeikis in your original example, you edited a cluster role binding which is not something customers are going to be able to do.

**Comment by mjudeikis** [comment]
ok, we might need to convert this to the story to provide customer `cluster-admin`, same as dedicates doing now. Where is custom cluster admin just elevated admin from multiple projects?
